unicorn/include/cocaine/unicorn/api.hpp

```diff
@@ -0,0 +1,111 @@
+/*
+* 2015+ Copyright (c) Anton Matveenko <antmat@yandex-team.ru>
+* All rights reserved.
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*/
+
+#ifndef COCAINE_UNICORN_API_HPP
+#define COCAINE_UNICORN_API_HPP
+
+#include "cocaine/unicorn/writable.hpp"
+
+namespace cocaine { namespace unicorn {
+class api_t {
```

Comment:
Why do you need a base class? Is it planned to add more than one successor of `unicorn::api_t`?